log_downloader_service.py

```diff
@@ -0,0 +1,87 @@
+import os
+import time
+import shutil
+from pathlib import Path
+from loguru import logger
+from git import Repo
+import wmi
+
+
+REPO_PATH = Path(os.path.dirname(os.path.dirname(os.path.abspath(__file__)))) # TODO: change
+LOGS_DIR = REPO_PATH / "logs"
+LOGS_DIR.mkdir(exist_ok=True)
+
+
+def commit_log(repo, log_file):
+    try:
+        repo.index.add([str(log_file)])
+        repo.index.commit(f"Add log file: {log_file.name}")
+        logger.info(f"Committed {log_file.name} to repository")
+    except Exception as e:
+        logger.info(f"Error committing {log_file.name}: {e}")
+
+
+def get_file_signature(file_path):
+    return file_path.name, file_path.stat().st_size
+
+
+def is_file_downloaded(log_file):
+    file_name = log_file.name
+    file_size = log_file.stat().st_size
+
+    for existing_file in LOGS_DIR.glob("*.wpilog"):
+        if (existing_file.name == file_name and
+                existing_file.stat().st_size == file_size):
+            return True
+    return False
+
+def get_usb_drives():
+    drives = set()
+    c = wmi.WMI()
+
+    for drive in c.Win32_LogicalDisk():
+        if drive.DriveType == 2: # removable
+            drives.add(Path(f"{drive.DeviceID}/"))
+
+    return drives
+
+def download_logs(drive_path, repo):
+    try:
+        for log_file in drive_path.glob("**/*.wpilog"):
```

Comment:
Perhaps we should download driver station logs too? What do you think?

Reply:
What do you mean by driverstation logs?

Reply:
The logs the Driver Station software collects itself (`.dsevents` file). These can be displayed on AdvntageScope as well.
But I would correct myself, don't download them from the robot, just copy the relevant log from the match from the Driver Station directory to our directory.

Reply:
5ff2ce4b9824dcb9f32d6fb7e3d3c904ec09c37e

Reply:
Again, this is something that shouldn't be handled in this function. It's only role is to download one log file. Finding it should be handled elsewhere.

Reply:
As we just discussed in the meeting, breaking code into smaller pieces, with each one having one responsibility, is good practice.